src/frontend/src/pages/TeamsPage/TeamMembersPageBlock.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Autocomplete, Button, Grid, IconButton, TextField, useTheme } from '@mui/material';
+import { useState } from 'react';
+import { useAuth } from '../../hooks/auth.hooks';
+import { useAllUsers } from '../../hooks/users.hooks';
+import { useSetTeamMembers } from '../../hooks/teams.hooks';
+import { Team, User } from 'shared';
+import { fullNamePipe } from '../../utils/pipes';
+import { Edit } from '@mui/icons-material';
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import PageBlock from '../../layouts/PageBlock';
+import DetailDisplay from '../../components/DetailDisplay';
+
+interface TeamMembersPageBlockProps {
+  team: Team;
+}
+
+const TeamMembersPageBlock: React.FC<TeamMembersPageBlockProps> = ({ team }) => {
+  const auth = useAuth();
+  const theme = useTheme();
+  const [isEditingMembers, setIsEditingMembers] = useState(false);
+  const [memberIds, setMemberIds] = useState(team.members.map((member) => member.userId));
+
+  const { isLoading, isError, error, data } = useAllUsers();
+  const {
+    isLoading: setTeamMemberIsLoading,
+    isError: setTeamMemberIsError,
+    error: setTeamMemberError,
+    mutateAsync
+  } = useSetTeamMembers(team.teamId);
+
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (setTeamMemberIsError) return <ErrorPage message={setTeamMemberError?.message} />;
+  if (isLoading || setTeamMemberIsLoading || !data) return <LoadingIndicator />;
+
+  const handleSubmit = async () => {
+    await mutateAsync(memberIds);
+    setIsEditingMembers(false);
+  };
+
+  const hasPerms = () => {
+    return !(
+      auth.user &&
+      (auth.user.role === 'ADMIN' || auth.user.role === 'APP_ADMIN' || auth.user.userId === team.leader.userId)
+    );
+  };
+
+  const userToAutocompleteOption = (user: User): { label: string; id: number } => {
+    return { label: `${fullNamePipe(user)}`, id: user.userId };
+  };
+
+  //first filters the options to only include users who have not been selected to be on the team, then sorts the options by alphabetical order of their first name, then maps the options to the autocomplete option format
+  const options = () => {
+    return data
+      .filter((user) => !memberIds.includes(user.userId))
+      .sort((a, b) => (a.firstName > b.firstName ? 1 : -1))
+      .map(userToAutocompleteOption);
+  };
+
+  const headerRight = (
+    <div style={{ display: 'flex' }}>
+      <Button onClick={() => setIsEditingMembers(false)}>Cancel</Button>
+      <Button
```

Comment:
this button looks good, we should maybe extract it to be like NERSuccessButton (ask @RChandler234 what he thinks)